main/end-to-end-tests/cypress/integration/appstream-egress-enabled/workspaces.spec.js

```diff
@@ -119,14 +122,29 @@ describe('Launch new workspaces', () => {
     expectedNumberOfNewlyOpenBrowserWindows += 1;
   }
 
-  function checkWorkspaceAvailableAndClickConnectionsButton(workspaceName) {
+  function checkRstudio(workspaceName) {
+    checkDetailsTable(workspaceName);
+    checkWorkspaceAvailableAndClickConnectionsButton(workspaceName);
     cy.contains(workspaceName)
       .parent()
-      .contains('AVAILABLE', { timeout: 900000 });
+      .find('[data-testid=sc-environment-generate-url-button]', { timeout: 60000 })
+      .click();
+
+    // Check we can not access Rstudio server url from the internet. Note rstudio server url is different from AppStream Url.
+    // To access Rstudio server we would do it through the AppStream Url
+    cy.get('[data-testid=destination-url]')
+      .invoke('text')
+      .then(url => {
+        cy.request({ url, failOnStatusCode: false }).then(response => {
+          expect(response.status).to.equal(403);
```

Comment:
Shouldn't this return a url not found error? I thought the RStudio Server is not accessible from the internet at all. In that case the server shouldn't return any response.

Reply:
This is the error:
<img width="619" alt="Screen Shot 2021-11-18 at 2 52 11 PM" src="https://user-images.githubusercontent.com/43092418/142487026-9e7e28f7-fa24-4d70-8159-b6d473051330.png">

Reply:
Tim and I worked together to make this test work. Changes coming soon.